crates/bevy_asset/src/io/mod.rs

```diff
@@ -27,20 +27,32 @@ use futures_lite::{ready, Stream};
 use std::{
     path::{Path, PathBuf},
     pin::Pin,
+    sync::Arc,
     task::Poll,
 };
 use thiserror::Error;
 
 /// Errors that occur while loading assets.
-#[derive(Error, Debug)]
+#[derive(Error, Debug, Clone)]
```

Comment:
Are we able to derive PartialEq here? If so, please do: it's really useful for writing tests on error types.

Reply:
Unfortunately not due to the `std::io::Error` in the `Io` variant